DeepProjection/pdbGenerator.py

```diff
@@ -36,36 +36,176 @@ def get_random_pdb(self):
+        3. The first character in the string for either lower or upper is a letter; not proper PDB ID format.
+        4. len(lower) or len(upper) is greater than 4; not proper PDB ID format
+        5. lower and upper contain upper case letters; just convert lower and upper to lower case.
+        6. Keep in Mind: wildcard symbol or a space is found in between two sets of letters (e.g. 1A*B, *BBB, " AA" or "AD C")
+        
+        Special Ranges
+        1. lower == upper; in this case, the function will attempt to download 1 file.
+        2. lower == '*' and upper == '*'; in this case, we will download all files.
+        3. lower is a valid ID, but upper == '*'; in this case, we will download all files between lower and upper == '9ZZZ'
+        4. lower == '*', but upper is a valid ID; in this case, we will download all files between lower == '0000' and upper
+        
+        """
+        
         alpha = ['a','b','c','d','e','f','g','h','i','j','k','l','m',
                  'n','o','p','q','r','s','t','u','v','w','x','y','z']
         num = ['0','1','2','3','4','5','6','7','8','9']
         
-        # Idea behind having the number go before the letters
-        # is to have 0000 be the starting ID, and go on from there.
+        # Idea behind having the numbers go before the letters
+        # is to count up from 0 to Z. So 0000 would be the first
+        # ID, 0001 would be the second, and so on
         numalpha = num + alpha
+
+        
+        
+        # Checking for Input Errors
+        if isinstance(dir, str) == False:
+            return False
+        elif (isinstance(lower, str) == False) or (isinstance(upper, str) == False):    # Input Error 1: lower and upper are not str.
+            return False
+        else:
+            # Input Error 5: lower and upper contain upper case letters; just convert lower and upper to lower case.
+            lower = lower.lower()
+            upper = upper.lower()
+
+
+        if (lower > upper) and (lower != '*') and (upper != '*'):   # Input Error 2: lower > upper (MAY BE SOURCE OF BUGS)
+            return False
+        elif (lower[0] in alpha) or (upper[0] in alpha):            # Input Error 3: The first character in either lower or upper is a letter
+            return False
+        elif (len(lower) > 4) or (len(upper) > 4):                  # Input Error 4: len(lower) or len(upper) is greater than 4; not proper PDB ID format
+            return False
+        else:
+            pass
         
-        list_of_valid_ids = []
-        currentID = '0000'
 
-        # Time to generate the codes
-        # The first position is always a number from 0 - 9
-        for first_position in range (10):
+        
+        # Checking for Special Ranges
+        # If lower == upper, our loop code should be able to handle that case on its own.
+
+        if lower == '*' and upper == '*': # Special Range 2: lower == '*' and upper == '*', so we will download all files
+            lower = '0000'
```

Comment:
First character in the PDB ID is between 1 and 9. Not zero.
In other characters, zero to 9.